cli/src/main/java/hudson/cli/CLI.java

```diff
@@ -189,8 +192,11 @@ private Channel connectViaCliPort(URL jenkins, CliPort clip) throws IOException
                 rsp.write(ch);
             }
             String head = new BufferedReader(new StringReader(rsp.toString("ISO-8859-1"))).readLine();
-            if (!head.startsWith("HTTP/1.0 200 "))
-                throw new IOException("Failed to establish a connection through HTTP proxy: "+rsp);
+            if (!(head.startsWith("HTTP/1.0 200 ") || head.startsWith("HTTP/1.1 200 "))) {
```

Comment:
Given HTTP 2.0 is a brand new protocol and that only HTTP 1.0 and HTTP 1.1 exist, I would prefer to keep the test as it is.